src/transformers/models/clip/modeling_clip.py

```diff
@@ -858,13 +874,29 @@ def forward(
         )
         return_dict = return_dict if return_dict is not None else self.config.use_return_dict
 
-        encoder_states = () if output_hidden_states else None
+        if isinstance(output_hidden_states, list):
+            output_hidden_states = [
+                min(self.config.num_hidden_layers, self.config.num_hidden_layers + index + 1) if index < 0 else index
+                for index in output_hidden_states
+            ]
```

Comment:
Before this, it would be better to validate the index range to make sure they are within the model's range, and raise if they are not. Then, the conversion should not be hidden in a `min` operation for clarity

Reply:
I've added validation of the index range.